src/serial.rs

```diff
@@ -319,21 +319,24 @@ macro_rules! hal {
                     // NOTE(unsafe) atomic read with no side effects
                     let isr = unsafe { (*$USARTX::ptr()).isr.read() };
 
+                    // NOTE(unsafe) write accessor for atomic writes with no side effects
```

Comment:
We had a discussion about using `write` in #36 .
Using it here makes sense, all I'm asking for is something like
```rust
// NOTE(unsafe, write) write accessor for atomic writes to a stateless register
```